### PR TITLE
Support Gemma3 with Clip fused attention

### PR DESCRIPTION
**Pull request by titaiwangms**
### Description

Essentially, the vision model is traced differently (this time it's without mask.), and the input indices of op.Add and op.MatMul can be different. Also, fp16 and fp32 need different tracing patterns (op.Cast).

1. Add another traced pattern to CLIP attention to cover no attention_mask case
2. Accept different index of input on op.Add and op.MatMul (be more general)
3. fp16 and fp32 shows different pattern (op.Cast after op.Softmax)
4. Refactor test_fastgelu.py to cover torch.onnx.export(..., dynamo=True)
5. Add gemma3 vision attention (SigLip) test to cover both fp16 and fp32

### Motivation and Context

To optimize Gemma3 multi-modal model, the changes are needed. https://huggingface.co/google/gemma-3-4b-it

### CODE REVIEW

File: onnxruntime/python/tools/transformers/fusion_attention_clip.py [hunk 1]
```diff
@@ -183,14 +193,27 @@ def fuse(self, normalize_node, input_name_to_nodes, output_name_to_node):
             if qk_nodes is None:
                 qk_nodes = self.model.match_parent_path(matmul_qkv, ["Softmax", "Add", "Mul", "MatMul"], [0, 0, 0, 0])
                 if qk_nodes is None:
-                    qk_nodes = self.model.match_parent_path(
-                        matmul_qkv, ["Cast", "Cast", "Softmax", "Add", "Mul", "MatMul"], [0, 0, 0, 0, 0, 0]
-                    )
+                    # If attention mask is not used, we can still match the qk path.
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/fusion_attention_clip.py:196]: Suggest to change the condition so that layout is more friendly:
Before:
```
if qk_nodes is None:
    ...
else:
   add_mask = qk_nodes[1]
```

To

```
if qk_nodes is not None:
   add_mask = qk_nodes[1]
else:
   ...
```

Another possible change is to use [match_parent_paths](https://github.com/microsoft/onnxruntime/blob/24620e70d9f14956a0dc84bb8a332dcd64c95a94/onnxruntime/python/tools/quantization/fusions/fusion.py#L276).

**Comment by titaiwangms** [comment]
The errors seem to be some docker image authorization issues.

**Comment by titaiwangms** [comment]
Closing this to "not using fork" for triggering CI (authorization to docker images)

**Comment by titaiwangms** [comment]
https://github.com/microsoft/onnxruntime/pull/24280
